lib/elixir/lib/exception.ex

```diff
@@ -1200,7 +1172,7 @@ defmodule ErlangError do
 
   def normalize({:badkey, key}, stacktrace) do
     term =
-      case ensure_stacktrace(stacktrace) do
+      case stacktrace do
         [{Map, :get_and_update!, [map, _, _], _} | _] -> map
         [{Map, :update!, [map, _, _], _} | _] -> map
```

Comment:
Do we really need the two clauses? I think we should make the functions in `Map` module raise full errors instead of examining the stacktrace afterwards.

Reply:
@michalmuskala so the idea was to make all of them consistent with Erlang/OTP. But I don't think we need to follow this rule. We can tackle it in a separate PR.